request.js

```diff
@@ -0,0 +1,45 @@
+var RequestModule = (function() {
+    // make an API request
+    function makeRequest(url) {
+        var xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function() {
+            if (xhr.readyState === 4 && xhr.status === 200) {
+                console.log(JSON.parse(xhr.responseText));
```

Comment:
might be worth putting the callback in the request module than we can call ```js
var response = JSON.parse(xhr.responseText));
callback(response);
```

And then we can easily understand what the callback function has passed to it